### PR TITLE
k8s: Move CiliumEnvoyConfig to v2

### PR DESCRIPTION
**Pull request by jrajahalme**
Move CiliumEnvoyConfig and CiliumClusterwideEnvoyConfig CRDs to API v2, as planned for Cilium release 1.12.

Note to reviewers: This PR should have no functional impact.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

### CODE REVIEW

File: pkg/k8s/apis/cilium.io/v2/logfields.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: Apache-2.0
 // Copyright Authors of Cilium
 
-package v2alpha1
+package v2
```
Comment (sayboras) [inline pkg/k8s/apis/cilium.io/v2/logfields.go:4]: we should change the version at line 11 in this file as well.

**Review by sayboras — APPROVED** [review]
one small comment as per below, the rest looks good to me.

For other reviewers, I have kicked off the conformance test for this PR, all tests are still good :100: :heavy_check_mark:

File: pkg/k8s/apis/cilium.io/v2/logfields.go (hunk 1 above)
Comment (jrajahalme) [reply pkg/k8s/apis/cilium.io/v2/logfields.go:4]: Done, thanks!

**Review by gandro — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]

**Review by christarazi — APPROVED** [review]

**Comment by sayboras** [comment]
the conformance test is added in this PR https://github.com/cilium/cilium/pull/19742

**Review by michi-covalent — APPROVED** [review]

**Comment by joestringer** [comment]
I just realized this is v2 and not v2alpha1. Don't we plan to change this arbitrarily as future Envoy configurations change? Do we just plan to bump to v3, v4, ... ?

**Pull request merged**